src/classes/UTIL_DMLService.cls

```diff
@@ -0,0 +1,306 @@
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToInsert The sObjects to add to the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> insertRecords(List<sObject> recordsToInsert, Database.DMLOptions dmlOptions) {
+        dmlOptions.optAllOrNone = ALL_OR_NONE_FALSE;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordsToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToInsert The sObject to add to the org.
+    * @param allOrNone The optional parameter that specifies whether the operation allows partial success.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult insertRecord(sObject recordToInsert, Boolean allOrNone, Database.DMLOptions dmlOptions) {
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToInsert The sObjects to add to the org.
+    * @param allOrNone The optional parameter that specifies whether the operation allows partial success.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> insertRecords(List<sObject> recordsToInsert, Boolean allOrNone, Database.DMLOptions dmlOptions) {
```

Comment:
why have both a version of insert that takes dmlOptions, and one that takes dmlOptions and a boolean that will get set into dmlOptions? Seems unnecessary and a bit confusing.

Reply:
Same as above.